OVERVIEW.md

```diff
@@ -0,0 +1,180 @@
+# Introduction and Overview
+
+## High-Level: What does it do and why do I need it?
+
+> TODO
+
+## Concepts
+
+* Andrew: Reuse and update "Introductions" section from [top level README](readme.md)
+
+
+### Built-in Rule Types
+
+The CloudDQ framework defines 5 basic types of rules:
+
+* NOT_NULL
+* NOT_BLANK
+* REGEX
+* CUSTOM_SQL_EXPRESSION
+* CUSTOM_SQL_STATEMENT
+
+The following table describes the details of these rules and how to use them.
+
+#### Rule Type `NOT_NULL`
+
+Violated if any row in the given column contains null.
+
+The rule is applied to a column in a table (an “entity”)  through a rule binding.
+
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `NOT_BLANK`
+
+Violated if any row in the given column contains null.
+
+The rule is applied to a column in a table (an “entity”)  through a rule binding.
+ 
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `REGEX`
+
+For `STRING` columns. This rule type allows the specification of a Python regular expression to define allowed patterns.
+
+The example shows a check for a valid email address.
+
+The rule is applied to a column in a table through a rule binding.
+
+```yaml
+rules:
+  VALID_EMAIL:
+    rule_type: REGEX
+    dimension: conformance
+    params:
+      pattern: |-
+        ^[^@]+[@]{1}[^@]+$
+
+rule_bindings:
+  VALID_EMAIL:
+    entity_id: TEST_DATA
+    column_id: EMAIL
+    row_filter_id: NONE
+    rule_ids:
+      - VALID_EMAIL
+```
+
+> Complete example TODO
+
+#### Rule Type `CUSTOM_SQL_EXPRESSION`
+
+This rule type allows the specification of a condition using SQL (as in a WHERE clause). The validation fails when the condition is violated.
```

Comment:
Suggest to make it clearer that: 1. CUSTOM_SQL EXPR supports row-level validations (btw pls specify this for all the previous rule types too); 2. It should be of type 'expression' returning a single boolean value indicating true when validation for the respective record passes, and false otherwise. This expression has access to the row context of the table defined as "entity" - for ex.all column values within the same record can be accessed